frontend/src/pages/LogsExplorer/__tests__/LogsExplorer.test.tsx

```diff
@@ -144,4 +159,102 @@ describe('Logs Explorer Tests', () => {
+					<I18nextProvider i18n={i18n}>
+						<MockQueryClientProvider>
+							<QueryBuilderContext.Provider
+								value={{
+									currentQuery: {
+										...initialQueriesMap.metrics,
+										builder: {
+											...initialQueriesMap.metrics.builder,
+											queryData: [
+												initialQueryBuilderFormValues,
+												initialQueryBuilderFormValues,
+											],
+										},
+									},
+									setSupersetQuery: jest.fn(),
+									supersetQuery: initialQueriesMap.metrics,
+									stagedQuery: initialQueriesMap.metrics,
+									initialDataSource: null,
+									panelType: PANEL_TYPES.TIME_SERIES,
+									isEnabledQuery: false,
+									handleSetQueryData: noop,
+									handleSetFormulaData: noop,
+									handleSetQueryItemData: noop,
+									handleSetConfig: noop,
+									removeQueryBuilderEntityByIndex: noop,
+									removeQueryTypeItemByIndex: noop,
+									addNewBuilderQuery: noop,
+									cloneQuery: noop,
+									addNewFormula: noop,
+									addNewQueryItem: noop,
+									redirectWithQueryBuilderData: noop,
+									handleRunQuery: noop,
+									resetQuery: noop,
+									updateAllQueriesOperators: (): Query => initialQueriesMap.metrics,
+									updateQueriesData: (): Query => initialQueriesMap.metrics,
+									initQueryBuilderData: noop,
+									handleOnUnitsChange: noop,
+									isStagedQueryUpdated: (): boolean => false,
+								}}
+							>
+								<VirtuosoMockContext.Provider
+									value={{ viewportHeight: 300, itemHeight: 100 }}
+								>
+									<LogsExplorer />
+								</VirtuosoMockContext.Provider>
+							</QueryBuilderContext.Provider>
+						</MockQueryClientProvider>
+					</I18nextProvider>
+				</Provider>
+			</MemoryRouter>,
+		);
+
+		const queries = queryAllByText(
+			'Search Filter : select options from suggested values, for IN/NOT IN operators - press "Enter" after selecting options',
+		);
+		expect(queries.length).toBe(2);
+
+		const legendFormats = queryAllByText('Legend Format');
+		expect(legendFormats.length).toBe(2);
+
+		const aggrInterval = queryAllByText('AGGREGATION INTERVAL');
+		expect(aggrInterval.length).toBe(2);
+	});
+
+	test('Logs Explorer Page - handle histogram visibility', async () => {
+		// mocking the query range API to return the logs
+		lodsQueryServerRequest();
+		const { container } = render(
+			<MemoryRouter initialEntries={[logExplorerRoute]}>
+				<Provider store={store}>
+					<I18nextProvider i18n={i18n}>
+						<MockQueryClientProvider>
+							<QueryBuilderProvider>
+								<VirtuosoMockContext.Provider
+									value={{ viewportHeight: 300, itemHeight: 100 }}
+								>
+									<LogsExplorer />
+								</VirtuosoMockContext.Provider>
+							</QueryBuilderProvider>
+						</MockQueryClientProvider>
+					</I18nextProvider>
+				</Provider>
+			</MemoryRouter>,
+		);
+
+		const histogramToggle = container.querySelector('button[role="switch"]');
+		expect(histogramToggle).toBeChecked();
```

Comment:
histogram toggle has already been tested in the first test of this file , can remove this

Reply:
Removed, thanks